web3/inpage.js

```diff
@@ -151,10 +151,33 @@ class Web3FrameProvider {
 
 if (!window.chrome || !window.chrome.extension) {
   console.log('Parity - Injecting Web3');
-  window.web3 = {
-    currentProvider: new Web3FrameProvider()
+
+  const web3 = {
+    currentProvider: new Web3FrameProvider(),
+    injectedWeb3: null
   };
 
+  const proxiedWeb3 = new Proxy(web3, {
```

Comment:
Right, I meant that the web3 dependency could be loaded lazily (but synchronously), sorry for miswrite :)